secure/crypto.go

```diff
@@ -0,0 +1,19 @@
+package secure
+
+import (
+	"errors"
+)
+
+func (s *SecureSession) encrypt(out, plaintext []byte) ([]byte, error) {
+	if s.enc == nil {
+		return nil, errors.New("cannot encrypt, handshake incomplete")
+	}
+	return s.enc.Encrypt(out, nil, plaintext), nil
+}
+
+func (s *SecureSession) decrypt(out, ciphertext []byte) ([]byte, error) {
```

Comment:
same as encrypto

Reply:
No handshake or handshake failure at any phase may cause this error. Cause of handshake failure is returned when using `NewSecureSession` to instantiate a `SecureSession`.

Same as above.

Reply:
error output changed